contrib/mesos/pkg/scheduler/podtask/procurement.go

```diff
@@ -17,31 +17,43 @@ limitations under the License.
 package podtask
 
 import (
+	"fmt"
+	"math"
+
 	log "github.com/golang/glog"
 	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesos/mesos-go/mesosutil"
+	"k8s.io/kubernetes/contrib/mesos/pkg/scheduler/executorinfo"
 	mresource "k8s.io/kubernetes/contrib/mesos/pkg/scheduler/resource"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
 )
 
 // NewDefaultProcurement returns the default procurement strategy that combines validation
 // and responsible Mesos resource procurement. c and m are resource quantities written into
 // k8s api.Pod.Spec's that don't declare resources (all containers in k8s-mesos require cpu
 // and memory limits).
-func NewDefaultProcurement(c mresource.CPUShares, m mresource.MegaBytes) Procurement {
-	resourceProcurer := &RequirePodResources{
-		defaultContainerCPULimit: c,
-		defaultContainerMemLimit: m,
-	}
+func NewDefaultProcurement(prototype *mesos.ExecutorInfo, eir executorinfo.Registry) Procurement {
 	return AllOrNothingProcurement([]Procurement{
-		ValidateProcurement,
-		NodeProcurement,
-		resourceProcurer.Procure,
-		PortsProcurement,
-	}).Procure
+		NewValidateProcurement(),
+		NewNodeProcurement(),
+		NewPodResourcesProcurement(),
+		NewPortsProcurement(),
+		NewExecutorResourceProcurer(prototype.GetResources(), eir),
```

Comment:
it'll be interesting to see how executor port resources will fit into this mold. not a problem that we have to solve here, just food for thought.